### PR TITLE
Add Kibana 5.1, 5.6, 6.0, 6.1 and 6.2.

### PR DESCRIPTION
**Pull request by UserNotFound**
Refactor from single Dockerfile based App, to direct image deploy method with multiple tags.

This App will use `before_release` to be sure that an Elasticsearch database of a supported version is reachable when starting.

The tests require the associated ES images to be available (ie quay.io/aptible/elasticsearch:6.1 must be tagged to test Kibana 6.1).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,64 +22,24 @@ should make sure you choose strong passwords for `AUTH_CREDENTIALS`.
...
 
-To run as an app on Aptible:
+To deploy Kibana as an App on Enclave:
 
- 1. Create an app in your [Aptible dashboard](https://dashboard.aptible.com) for Kibana. In the
-    steps that follow, we'll use &lt;YOUR_KIBANA_APP_HANDLE&gt; anywhere that you should substitute the
-    actual app handle the results from this step in the instructions.
+   aptible apps:create $HANDLE 
+   aptible config:set --app $HANDLE AUTH_CREDENTIALS=username:password DATABASE_URL=http://user:password@example.com
+   aptible deploy --app $HANDLE --docker-image quay.io/aptible/kibana
```
Comment (krallin) [inline README.md]: A few requests here:

- Merge config:set and deploy into a single call
- Quote variables
- Use `aptible/kibana` rather than the `quay.io` repo.

File: bin/check-es-version.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+#shellcheck disable=SC2086
+
+
+DATABASE_URL=$1
+WAIT=10
+
+function wait_for_reqeust {
```
Comment (krallin) [inline bin/check-es-version.sh]: There's a typo here (`wait_for_reqeust` -> `wait_for_request`).

File: bin/check-es-version.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    sleep 1
+  done
+
+  echo "Unable to reach Elasticsearch server, please check DATABASE_URL and your database server."
+  exit 1
+}
+
+
+# The old, Dockerfile based deploy won't have $TAG set
+if [[ -z "$TAG" ]]; then
```
Comment (krallin) [inline bin/check-es-version.sh]: How exactly is someone going to get here? Can you clarify the thinking here?

I think our original intention was to retain a minimal Dockerfile in the repository that simply prints this error when built (and exits, thus failing the build), and invites the user to either update to the new method, or check out an older commit.

File: bin/check-es-version.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  print 4.1 if es_version.start_with?('1.')
+  print 4.4 if es_version.start_with?('2.')
+  print 5.0 if es_version.start_with?('5.0.')
+  print 5.1 if es_version.start_with?('5.1.')
+  print 6.0 if es_version.start_with?('6.0.')
+  print 6.1 if es_version.start_with?('6.1.')"
+
+  wait_for_reqeust
+
+  KIBANA_NEEDED_VERSION="$(curl "$DATABASE_URL" 2>/dev/null | ruby -e "$KIBANA_VERSION_PARSER")"
```
Comment (krallin) [inline bin/check-es-version.sh]: Consider using `curl -fsSL` and remove your stderr redirection - that way you'll get an error if something goes wrong with curl, but you won't get anything if not.

File: bin/check-es-version.sh [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+#shellcheck disable=SC2086
```
Comment (krallin) [inline bin/check-es-version.sh:2]: Consider enabling:

- pipefail to know if curl fails when you pipe it to the parser.
- nounset because you realistically should not be accessing unset variables here.
- errexit because you want to error if e.g. curl fails.

File: bin/check-es-version.sh [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+function wait_for_reqeust { 
+  for _ in $(seq 1 ${WAIT}); do
+    if ! curl $"DATABASE_URL" > /dev/null 2>&1; then
+      return 0
+    fi
+    sleep 1
+  done
+
+  echo "Unable to reach Elasticsearch server, please check DATABASE_URL and your database server."
+  exit 1
```
Comment (krallin) [inline bin/check-es-version.sh]: With `errexit` you can make this more reusable with `return 1`.

File: bin/check-es-version.sh [hunk 7]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+#shellcheck disable=SC2086
+
+
+DATABASE_URL=$1
+WAIT=10
+
+function wait_for_reqeust { 
+  for _ in $(seq 1 ${WAIT}); do
```
Comment (krallin) [inline bin/check-es-version.sh]: `${WAIT}` -> `"$WAIT"`

File: bin/check-es-version.sh [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  wait_for_reqeust
+
+  KIBANA_NEEDED_VERSION="$(curl "$DATABASE_URL" 2>/dev/null | ruby -e "$KIBANA_VERSION_PARSER")" 
+
+  if [[ -z "$KIBANA_NEEDED_VERSION" ]]; then
+    echo "Your version of Elasticsearch not supported by this application."
+    exit 1
+  fi
+ 
+  if [[ $TAG != $KIBANA_NEEDED_VERSION ]]; then
```
Comment (krallin) [inline bin/check-es-version.sh]: Quotes

File: bin/check-es-version.sh [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    echo "You're using docker-kibana:${TAG}. To connect to this version of Elasticsearch,"
```
Comment (krallin) [inline bin/check-es-version.sh]: The image is not called `docker-kibana`, so this is going to be confusing. Rename this to `aptible/kibana:${TAG}`.

Besides, I would suggest rewording this to indicate that the reason they need to update to connect is because Elasticsearch is not compatible with that version.

File: bin/check-es-version.sh [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    echo "you need to use docker-kibana:${KIBANA_NEEDED_VERSION}"
```
Comment (krallin) [inline bin/check-es-version.sh]: Same as above

File: bin/run-kibana.sh [hunk 11]
```diff
@@ -24,34 +24,21 @@ erb -T 2 -r uri -r base64 ./kibana.erb > /etc/nginx/sites-enabled/kibana || {
...
 # Run config
-erb -T 2 -r uri "/opt/kibana-${KIBANA_VERSION}/config/kibana.yml.erb" > "/opt/kibana-${KIBANA_VERSION}/config/kibana.yml" || {
+erb -T 2 -r uri -r base64 "/opt/kibana/config/kibana.yml.erb" > "/opt/kibana/config/kibana.yml" || {
   echo "Error creating kibana config file"
   exit 1
 }
 
+# Don't bother checking ES version before startup in testing, some tests don't actually start ES
+if [[ -z "$TESTING" ]]; then
+  bash ./check-es-version.sh || {
```
Comment (krallin) [inline bin/run-kibana.sh]: - Ensure the script is executable and has a shebang, and remove the `bash` call here
- I'm not a fan of expecting `check-es-version.sh` to be in the current directory 

Besides, consider using `if ! ...` as opposed to `|| { }` to make the intent a little clearer here

File: test-auth.sh [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+KIBANA_CREDS=kuser:kpass
+ES_USER=euser 
+ES_PASS=epass
+
+function cleanup {
+  echo "Cleaning up"
+  docker rm -f "$DB_CONTAINER" "$DATA_CONTAINER" "$KIBANA_CONTAINER" > /dev/null 2>&1 || true
+}
+
+function wait_for_reqeust {
```
Comment (krallin) [inline test-auth.sh]: Typo in the name here

File: test-auth.sh [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  CONTAINER=$1
+  REQUEST=$2
```
Comment (krallin) [inline test-auth.sh]: Quotes everywhere

File: test.sh [hunk 14]
```diff
@@ -7,4 +7,6 @@ IMG="$REGISTRY/$REPOSITORY:$TAG"
 echo "Running bats tests"
 docker run -i --entrypoint "bats" "$IMG" "/tmp/test"
 
+./test-auth.sh ${TAG} ${IMG}
```
Comment (krallin) [inline test.sh]: - Quotes
- Consider reverting `TAG` and `IMG` here (more specific last, least specific first)

File: test-auth.sh [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+
+echo "Verifying DB shutdown message isn't present"
+docker logs "$DB_CONTAINER" 2>&1 | grep -vqiE "node.+closed"
+
+echo "Sarting Kibana"
+docker run -d --name="$KIBANA_CONTAINER" \
+  -e DATABASE_URL="http://${ES_USER}:${ES_PASS}@${ES_IP}:80" \
+  -e AUTH_CREDENTIALS="$KIBANA_CREDS" \
+  -e TESTING="true" \
+  -p 5601:5601 \
```
Comment (krallin) [inline test-auth.sh]: Not a fan of using a specific port here - consider `docker exec`'ing into the Kibana container

File: test-auth.sh [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+  "$IMG" 
+
+#sleep 600
```
Comment (krallin) [inline test-auth.sh]: Remove this if not used

File: Dockerfile.erb [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+ADD ${TAG}/templates/kibana.yml.erb /opt/kibana/config/
+
+<% if ENV["TAG"] =="4.4" %>
+ADD ${TAG}/patches /patches
+RUN patch -p1 -d "/opt/kibana" < /patches/0001-Set-authorization-header-when-connecting-to-ES.patch
+<% end %>
+
+# Add script that starts NGiNX in front of Kibana and tails the NGiNX access/error logs.
+ADD bin .
+ADD .aptible.yml .
```
Comment (krallin) [inline Dockerfile.erb]: This is not the correct path for embedding - that would be `/.aptible/.aptible.yml`

File: Dockerfile.erb [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+RUN chmod 700 ./run-kibana.sh
+RUN chmod 700 ./check-es-version.sh
+
+# Add tests. Those won't run as part of the build because customers don't need to run
+# them when deploying, but they'll be run in test.sh
+ADD test /tmp/test
+ADD ${TAG}/test /tmp/test
+
+EXPOSE 80
+EXPOSE 5601
```
Comment (krallin) [inline Dockerfile.erb]: We use to expose 2 ports, now we're only exposing one — how should customers choose which port to expose?

**Review by krallin — CHANGES_REQUESTED** [review]
Thanks!

I left a number of comments here - these are mostly simple changes. Can you take a look at hose?

I haven't tried running this just yet, but I will do so tomorrow (along with your Elasticsearch changes).

**Comment by krallin** [comment]
Also: you'll need to update `latest.mk` to reference `6.1` as the latest tag here.

File: bin/check-es-version.sh (hunk 5 above)
Comment (UserNotFound) [reply bin/check-es-version.sh:2]: Addressed, but on a line not marked noted here to make it outdated.

File: bin/check-es-version.sh [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+print 4.1 if es_version.start_with?('1.')
+print 4.4 if es_version.start_with?('2.')
+print 5.0 if es_version.start_with?('5.0.')
+print 5.1 if es_version.start_with?('5.1.')
+print 6.0 if es_version.start_with?('6.0.')
+print 6.1 if es_version.start_with?('6.1.')"
+
+wait_for_request
+
+sleep 1
```
Comment (krallin) [inline bin/check-es-version.sh]: Why is this needed?

File: 4.4/config.mk [hunk 20]
```diff
@@ -0,0 +1,3 @@
+export KIBANA_VERSION=4.4.2
+export KIBANA_DOWNLOAD=https://download.elastic.co/kibana/kibana/kibana-${KIBANA_VERSION}-linux-x64.tar.gz"
```
Comment (krallin) [inline 4.4/config.mk]: There is a trailing quote here (this is causing your tests to crash).

File: 5.1/config.mk [hunk 21]
```diff
@@ -0,0 +1,3 @@
+export KIBANA_VERSION="5.1.2"
+export KIBANA_DOWNLOAD="https://artifacts.elastic.co/downloads/kibana/kibana-${KIBANA_VERSION}-linux-x86_64.tar.gz"
```
Comment (krallin) [inline 5.1/config.mk]: Remove the quotes here: this is a makefile (not a shell script), so quotes are interpreted as text.

Also, the syntax for interpolation in a Makefile is `$(KIBANA_VERSION)`, not `${KIBANA_VERSION}`.

Overall, this should be:

```
export KIBANA_VERSION = 5.1.2
export KIBANA_DOWNLOAD = https://artifacts.elastic.co/downloads/kibana/kibana-$(KIBANA_VERSION)-linux-x86_64.tar.gz
export KIBANA_SHA1SUM = ba355d63fef6702109ebdbf72d7ebca0451ed7ae
```

File: 6.0/config.mk [hunk 22]
```diff
@@ -0,0 +1,3 @@
+export KIBANA_VERSION="6.0.1"
+export KIBANA_DOWNLOAD="https://artifacts.elastic.co/downloads/kibana/kibana-${KIBANA_VERSION}-linux-x86_64.tar.gz"
+export KIBANA_SHA1SUM="89176814fb2e293ef03d542f707767bf21397d6a"
```
Comment (krallin) [inline 6.0/config.mk]: More quotes to remove

File: 6.1/config.mk [hunk 23]
```diff
@@ -0,0 +1,3 @@
+export KIBANA_VERSION="6.1.3"
+export KIBANA_DOWNLOAD="https://artifacts.elastic.co/downloads/kibana/kibana-${KIBANA_VERSION}-linux-x86_64.tar.gz"
+export KIBANA_SHA1SUM="9340dd0f71923e6fe517fc0c84a455602216b70e"
```
Comment (krallin) [inline 6.1/config.mk]: More quotes to remove and interpolation to fix.

File: test-auth.sh [hunk 24]
```diff
@@ -0,0 +1,80 @@
...
+  >/dev/null 2>&1
+
+docker run -d --name="$DB_CONTAINER" \
+  --volumes-from "$DATA_CONTAINER" \
+  "${ES_IMG}"
+
+echo "Waiting for DB to come online"
+wait_for_request "$DB_CONTAINER" "http://localhost:9200"
+
+ES_IP=$(docker inspect ${DB_CONTAINER} | grep '\"IPAddress\"' | head -n 1 | awk -F "\"" '{print $4}')
```
Comment (krallin) [inline test-auth.sh]: Don't try and parse JSON using grep / head / awk. Instead, pass a format to `docker inspect` 

```
ES_IP="$(docker inspect --format '{{ .NetworkSettings.IPAddress }}' "$DB_CONTAINER")"
```

File: test-auth.sh [hunk 25]
```diff
@@ -0,0 +1,80 @@
...
+
+echo "Verifying DB shutdown message isn't present"
+docker logs "$DB_CONTAINER" 2>&1 | grep -vqiE "node.+closed"
```
Comment (krallin) [inline test-auth.sh]: Why is this relevant here?

File: 4.1/config.mk [hunk 26]
```diff
@@ -0,0 +1,3 @@
+export KIBANA_VERSION=4.1.11
+export KIBANA_DOWNLOAD=https://download.elastic.co/kibana/kibana/kibana-${KIBANA_VERSION}-linux-x64.tar.gz
```
Comment (krallin) [inline 4.1/config.mk]: Fix the interpolation here

File: 5.0/config.mk [hunk 27]
```diff
@@ -0,0 +1,3 @@
+export KIBANA_VERSION=5.0.1
+export KIBANA_DOWNLOAD=https://artifacts.elastic.co/downloads/kibana/kibana-${KIBANA_VERSION}-linux-x86_64.tar.gz
```
Comment (krallin) [inline 5.0/config.mk]: Fix the interpolation here

File: test-auth.sh [hunk 28]
```diff
@@ -0,0 +1,80 @@
...
+
+echo "Sarting Kibana"
+docker run -d --name="$KIBANA_CONTAINER" \
+  -e DATABASE_URL="http://${ES_USER}:${ES_PASS}@${ES_IP}:80" \
+  -e AUTH_CREDENTIALS="$KIBANA_CREDS" \
+  -e TESTING="true" \
+  "$IMG" 
+
+wait_for_request "${KIBANA_CONTAINER}" "http://localhost:5601/elasticsearch/"
```
Comment (krallin) [inline test-auth.sh]: Does this actually test anything? On 6.1  at the very least, it passes even if I comment out the auth headers required to talk to ES (this URL simply returns a redirect).

I looked into the requests made by Kibana to ES. Here is a minimal test that actually does something useful. This appears to work on >= 4.4 (can you try and add another one that works for 4.1?):

```
wait_for_request "${KIBANA_CONTAINER}" \
  'http://localhost:5601/elasticsearch/*/_search' \
  -H "kbn-version: ${KIBANA_VERSION}" \
  -H 'content-type: application/json' \
  --data-binary '{}'
```

To make this work, you need `wait_for_request` to accept `$@`, like this:

```
function wait_for_request {
  CONTAINER=$1
  for _ in $(seq 1 30); do
    if docker exec -it "$CONTAINER" curl -f -v "$@"; then
      return 0
    fi
    sleep 1
  done

  echo "No response"
  docker logs "$CONTAINER"
  return 1
}
```

You also  need to pass `KIBANA_VERSION` to this test.

File: test.sh [hunk 29]
```diff
@@ -7,4 +7,6 @@ IMG="$REGISTRY/$REPOSITORY:$TAG"
 echo "Running bats tests"
 docker run -i --entrypoint "bats" "$IMG" "/tmp/test"
 
+./test-auth.sh ${IMG} ${TAG}
```
Comment (krallin) [inline test.sh]: Can this gets quotes? This is a trivial case that gets called out by Shellcheck.

Can you please make sure it is installed and integrated with whatever editor you are using here?

**Review by krallin — CHANGES_REQUESTED** [review]

File: test-auth.sh (hunk 25 above)
Comment (UserNotFound) [reply test-auth.sh]: It's not needed by this test; it simply helped debug an ES issue when writing it and will be removed.

File: bin/check-es-version.sh (hunk 19 above)
Comment (UserNotFound) [reply bin/check-es-version.sh]: The sleep "fixes" a race condition in testing, where the `wait_for_request` curl was satisfied by the `echo | nc` loop, but the following `curl` where we try to grab the response body would fail.

```
( while echo '{"version": {"number": "0.8"}}' | nc -l 456; do : ; done ) &
/bin/bash check-es-version.sh http://localhost:456
```

If we should use more reliable way to simulate/provide a response, I'd be happy to implement it.

File: README.md [hunk 30]
```diff
@@ -22,64 +22,23 @@ should make sure you choose strong passwords for `AUTH_CREDENTIALS`.
...
 ## Installation
 
-To run as an app on Aptible:
+To deploy Kibana as an App on Enclave:
 
- 1. Create an app in your [Aptible dashboard](https://dashboard.aptible.com) for Kibana. In the
-    steps that follow, we'll use &lt;YOUR_KIBANA_APP_HANDLE&gt; anywhere that you should substitute the
-    actual app handle the results from this step in the instructions.
+   aptible apps:create "$HANDLE" 
+   aptible deploy --app "$HANDLE" --docker-image aptible/kibana AUTH_CREDENTIALS=username:password DATABASE_URL=http://user:password@example.com
```
Comment (krallin) [inline README.md]: There are many usability problems here:

- This is completely unreadable once rendered to Markdown. You need to add code fences (i.e. triple backquotes) above and below these two lines to ensure your newlines are interpreted as newlines (and to get code formatting). Consider using the rich view of the diff in Github once you've made changes in order to confirm this looks like what you expect (if you don't know how to do that, let me know, and I'll show you).
- Can we explain what `$HANDLE` is, and describe what these commands do? 
- Can we explain what `AUTH_CREDENTIALS` do?
- Can we wrap `AUTH_CREDENTIALS=...` in quotes for users that might (intentionally or not) use a shell-reserved keyword in their credentials?
- Can we wrap `DATABASE_URL` in quotes (largely for consistency, but the same reasons apply)
- Can we explain what `DATABASE_URL` is, and use a  `DATABASE_URL` that remotely resembles the one an Aptible user is probably going to use? This repo is mainly designed to be used by Enclave users: we should simply tell them that `DATABASE_URL` is their connection URL rather than expect them to figure that out by indicating that their database should be "reachable there".

File: README.md [hunk 31]
```diff
@@ -22,64 +22,23 @@ should make sure you choose strong passwords for `AUTH_CREDENTIALS`.
...
 
- 2. Use the [Aptible CLI](https://github.com/aptible/aptible-cli) to set AUTH_CREDENTIALS to the
-    username/password you want to use to access the app. To set the user to "foo" and password
-    to "bar", run:
+Note: your Elasticsearch database must be reachable at `DATABASE_URL` when this image is deployed, and the Kibana version must be compatible with the Elasticsearch version you are connecting to.
```
Comment (krallin) [inline README.md]: Telling the user *what* should be is useful, but telling them *how* to do it is more useful:

- For the URL, tell them it's their Aptible connection URL(I noted this above)
- For the compatibility issue, tell them to look at the list below.

File: bin/check-es-version.sh (hunk 5 above)
Comment (krallin) [reply bin/check-es-version.sh:2]: Thanks!

File: test-auth.sh [hunk 32]
```diff
@@ -0,0 +1,81 @@
...
+  -e DATABASE_URL="http://${ES_USER}:${ES_PASS}@${ES_IP}:80" \
+  -e AUTH_CREDENTIALS="$KIBANA_CREDS" \
+  -e TESTING="true" \
+  "$IMG" 
+
+wait_for_request "${KIBANA_CONTAINER}" \
+  'http://localhost:5601/elasticsearch/*/_search' \
+  -H "kbn-version: $KIBANA_VERSION" \
+  -H 'content-type: application/json' \
+  --data-binary '{}'
```
Comment (krallin) [inline test-auth.sh:81]: :+1:

File: test-auth.sh [hunk 33]
```diff
@@ -0,0 +1,81 @@
...
+docker run -d --name="$DB_CONTAINER" \
+  --volumes-from "$DATA_CONTAINER" \
+  "${ES_IMG}"
+
+echo "Waiting for DB to come online"
+wait_for_request "$DB_CONTAINER" "http://localhost:9200"
+
+ES_IP=$(docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' ${DB_CONTAINER})
+
+echo "Sarting Kibana"
```
Comment (krallin) [inline test-auth.sh]: Sarting -> Starting

File: test-auth.sh [hunk 34]
```diff
@@ -0,0 +1,81 @@
...
+  >/dev/null 2>&1
+
+docker run -d --name="$DB_CONTAINER" \
+  --volumes-from "$DATA_CONTAINER" \
+  "${ES_IMG}"
+
+echo "Waiting for DB to come online"
+wait_for_request "$DB_CONTAINER" "http://localhost:9200"
+
+ES_IP=$(docker inspect --format='{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' ${DB_CONTAINER})
```
Comment (krallin) [inline test-auth.sh]: This is a nit, but ...

- I'm not certain we need to account for multiple networks here (but note that you might want to specifically ask for the bridge: `docker inspect --format "{{.NetworkSettings.Networks.bridge.IPAddress}}" "$DB_CONTAINER"`.
- I gave you this in an earlier comment, and you have functionally the same thing, except you dropped all quoting — is there a reason why?

File: test/kibana.bats [hunk 35]
```diff
@@ -56,125 +49,46 @@ teardown() {
...
-  done
-
-  pkill node
-  pkill nc
-
-  # Check that a authorization header was sent to "Elasticsearch" for the right
-  # credentials.
-  grep -A8 ".kibana/" "$web_log" | grep -i "Authorization: Basic dXNlcjpwYXNz"
-}
+@test "docker-kibana etects inability to connect to Elasticsearch" {
```
Comment (krallin) [inline test/kibana.bats]: etects -> detects

**Review by krallin — CHANGES_REQUESTED** [review]

File: 6.2/test/kibana-6.1.bats [hunk 36]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bats
```
Comment (krallin) [inline 6.2/test/kibana-6.1.bats]: (nit) presumably you meant to call this file `6.2/test/kibana-6.2.bats`, not `6.2/test/kibana-6.1.bats`.

File: README.md [hunk 37]
```diff
@@ -19,72 +19,41 @@ against the upstream Elasticsearch instance using Kibana's credentials**.
 This is probably what you want if you're deploying Kibana, but it means you
```
Comment (krallin) [inline README.md:17]: README one paragraph up (out of the diff, so I can't comment there) says:

>  Kibana as an Aptible app. This app automatically detects your Elasticsearch
> version and starts Kibana 4.1, 4.4, or 5.0 accordingly.

Can you update accordingly? (the latter is no longer correct)

File: README.md [hunk 38]
```diff
@@ -19,72 +19,41 @@ against the upstream Elasticsearch instance using Kibana's credentials**.
...
-
- 1. Create an app in your [Aptible dashboard](https://dashboard.aptible.com) for Kibana. In the
-    steps that follow, we'll use &lt;YOUR_KIBANA_APP_HANDLE&gt; anywhere that you should substitute the
-    actual app handle the results from this step in the instructions.
+To deploy Kibana as an App on Enclave:
 
- 2. Use the [Aptible CLI](https://github.com/aptible/aptible-cli) to set AUTH_CREDENTIALS to the
-    username/password you want to use to access the app. To set the user to "foo" and password
-    to "bar", run:
+1. Create a new App for Kibana. In the step that follows, we'll use `$HANDLE` anywhere that you should substitute the actual App handle the results from this step.
```
Comment (krallin) [inline README.md]: > we'll use `$HANDLE` anywhere that you should substitute the actual App handle the results from this step.

I think there is something wrong with this sentence - can you fix?

File: README.md [hunk 39]
```diff
@@ -19,72 +19,41 @@ against the upstream Elasticsearch instance using Kibana's credentials**.
...
- 6. Clone this repository and push it to your Aptible app:
-
-    ```
-    git clone https://github.com/aptible/docker-kibana.git
-    cd docker-kibana
-    git remote add aptible git@beta.aptible.com:<YOUR_KIBANA_APP_HANDLE>.git
-    git push aptible master
-    ```
+* `latest`: Currently Kibana 6.1
+* `6.1`: Kibana 6.1.3 (for Elasticsearch 6.1.x)
```
Comment (krallin) [inline README.md]: Add 6.2 here?

File: latest.mk [hunk 40]
```diff
@@ -1,4 +1 @@
-# There isn't really a "latest" TAG for this repository (there's
-# only one variatio and should ideally ever only be one), so we
-# just use latest here.
-LATEST_TAG = latest
+LATEST_TAG = 6.1
```
Comment (krallin) [inline latest.mk]: Make this 6.2

File: README.md [hunk 41]
```diff
@@ -19,72 +19,41 @@ against the upstream Elasticsearch instance using Kibana's credentials**.
...
-
- 6. Clone this repository and push it to your Aptible app:
-
-    ```
-    git clone https://github.com/aptible/docker-kibana.git
-    cd docker-kibana
-    git remote add aptible git@beta.aptible.com:<YOUR_KIBANA_APP_HANDLE>.git
-    git push aptible master
-    ```
+* `latest`: Currently Kibana 6.1
```
Comment (krallin) [inline README.md]: Update to 6.2?

**Review by krallin — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 42]
```diff
@@ -6,7 +6,12 @@ services:
 
 env:
   matrix:
-    - TAG=latest
+    - TAG=4.1
+    - TAG=4.4
+    - TAG=5.0
+    - TAG=5.1
+    - TAG=6.0
+    - TAG=6.1
```
Comment (krallin) [inline .travis.yml]: Add 6.2 here so that it actually builds.

**Review by krallin — APPROVED** [review]
Looks good to go; thanks!

**Comment by krallin** [comment]
That said, can you merge https://github.com/aptible/docker-elasticsearch/pull/42 then re-run the tests here that depend on it, just to be safe?
